File: web/app/Models/Event.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Event extends Model
{
    public const CATEGORIES = [
        'Música' => ['Concierto', 'Festival', 'DJ / Electrónica', 'Otros'],
        'Deportes' => ['Fútbol', 'Running', 'Fitness', 'Otros'],
        'Negocios' => ['Networking', 'Conferencia', 'Taller', 'Otros'],
        'Arte y cultura' => ['Teatro', 'Exposición', 'Cine', 'Otros'],
        'Gastronomía' => ['Cata', 'Feria', 'Clase de cocina', 'Otros'],
        'Comunidad' => ['Charla', 'Voluntariado', 'Encuentro social', 'Otros'],
        'Otros' => ['General'],
    ];

    protected $fillable = [
        'user_id',
        'organization_id',
        'title',
        'slug',
        'description',
        'category',
        'subcategory',
        'cover_image',
        'address',
        'latitude',
        'longitude',
        'start_at',
        'end_at',
        'is_public',
        'status',
    ];

    protected function casts(): array
    {
        return [
            'start_at' => 'datetime',
            'end_at' => 'datetime',
            'is_public' => 'boolean',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function organization(): BelongsTo
    {
        return $this->belongsTo(Organization::class);
    }

    public function ticketTypes(): HasMany
    {
        return $this->hasMany(TicketType::class);
    }

    public function orders(): HasMany
    {
        return $this->hasMany(Order::class);
    }

    public function tickets(): HasMany
    {
        return $this->hasMany(Ticket::class);
    }

    public function savedByUsers(): BelongsToMany
    {
        return $this->belongsToMany(User::class, 'saved_events')->withTimestamps();
    }

    public function revenue(): float
    {
        return (float) $this->orders()->where('status', 'paid')->sum('total');
    }
}

File: apps/attendee/lib/models/event.dart
import 'ticket_type.dart';

class Event {
  final int id;
  final String title;
  final String slug;
  final String? description;
  final String? category;
  final String? subcategory;
  final String? coverImage;
  final String? address;
  final DateTime startAt;
  final DateTime? endAt;
  final bool isSaved;
  final List<TicketType> ticketTypes;

  Event({
    required this.id,
    required this.title,
    required this.slug,
    this.description,
    this.category,
    this.subcategory,
    this.coverImage,
    this.address,
    required this.startAt,
    this.endAt,
    this.isSaved = false,
    this.ticketTypes = const [],
  });

  factory Event.fromJson(Map<String, dynamic> json) => Event(
        id: json['id'] as int,
        title: json['title'] as String,
        slug: json['slug'] as String,
        description: json['description'] as String?,
        category: json['category'] as String?,
        subcategory: json['subcategory'] as String?,
        coverImage: json['cover_image'] as String?,
        address: json['address'] as String?,
        startAt: DateTime.parse(json['start_at'] as String),
        endAt: json['end_at'] != null ? DateTime.parse(json['end_at'] as String) : null,
        isSaved: json['is_saved'] as bool? ?? false,
        ticketTypes: (json['ticket_types'] as List<dynamic>?)
                ?.map((e) => TicketType.fromJson(e as Map<String, dynamic>))
                .toList() ??
            const [],
      );
}
